model/project_parser.go

```diff
@@ -685,6 +690,34 @@ func retrieveFile(ctx context.Context, opts GetProjectOpts) ([]byte, error) {
 	}
 }
 
+func retrieveFileForModule(ctx context.Context, opts GetProjectOpts, modules ModuleList, moduleName string) ([]byte, error) {
```

Comment:
Right now this does have a lot of copied code from retrieveFile, which I don't think is necessary. If we change GetProjectOpts to accept Owner and Repo strings (instead of Ref) then `retrieveFileForModule` can just set owner/repo and use retrieveFile, which would save you from repeating 699 - 719

Reply:
everything uses retrieveFile